store/src/java/com/zimbra/cs/html/owasp/OwaspPolicy.java

```diff
@@ -0,0 +1,153 @@
+
+    static {
+        try {
+            load(null);
+        } catch (DocumentException de) {
+            throw new RuntimeException(de);
+        } catch (Exception ce) {
+            throw new RuntimeException(ce);
+        }
+    }
+
+    private OwaspPolicy(String file) throws DocumentException, Exception {
+        mPolicyFile = file;
+        if (mPolicyFile == null) {
+            mPolicyFile = defaultPolicyFile();
+        }
+        File policyFile = new File(mPolicyFile);
+        if (policyFile.exists() && policyFile.canRead()) {
+            try (FileInputStream fis = new FileInputStream(policyFile)) {
+                Document document = W3cDomUtil.parseXMLToDom4jDocUsingSecureProcessing(fis);
+                Element root = document.getRootElement();
+                if (!root.getName().equals(E_OWASP_POLICY)) {
+                    ZimbraLog.mailbox
+                        .warn(String.format("OWASP policy file '%s' root tag is not '%s'",
+                            mPolicyFile, E_OWASP_POLICY));
+                    throw new DocumentException(
+                        String.format("OWASP policy file '%s' root tag is not '%s'", mPolicyFile, E_OWASP_POLICY));
+                }
+                for (Iterator<?> iter = root.elementIterator(E_HTML); iter.hasNext();) {
+                    Element html = (Element) iter.next();
+                    String element = html.attributeValue(A_ELEMENT);
+                        String attributes = html.elementText(E_ATTRIBUTES);
+                        attributes = attributes.replace("CORE", "id,class,title,style")
+                            .replace("LANG", "dir,lang,xml:lang")
+                            .replace("KBD", "accesshtml,tabindex");
+                        set(element, attributes);
+                }
+                String disallowTextElements = root.elementText(E_DISALLOW_TEXT_IN);
+                mDisallowTextElements.addAll(Arrays.asList(disallowTextElements.split(COMMA)));
+                String urlProtocols = root.elementText(E_URL_PROTOCOLS);
+                mURLProtocols.addAll(Arrays.asList(urlProtocols.split(COMMA)));
+            } catch (IOException | XmlParseException e) {
+                ZimbraLog.mailbox
+                    .warn(String.format("Problem parsing owasp policy file '%s'", policyFile), e);
+                throw new DocumentException(
+                    String.format("Problem parsing owasp policy file '%s'", policyFile));
+            }
+        } else {
+            ZimbraLog.mailbox
+                .warn(String.format("Owasp policy file '%s' is not readable", mPolicyFile));
+            throw new Exception(
+                String.format("Owasp policy file '%s' is not readable", mPolicyFile));
+        }
+    }
+
+    private String defaultPolicyFile() {
+        final String FS = File.separator;
+        return FS + "opt" + FS + "zimbra" + FS + "conf" + FS + "owasp_policy.xml";
```

Comment:
There is a LocalConfig variable zimbra_home which I suspect we should be using here instead of assuming /opt/zimbra - although I doubt anyone has tried to use a different directory in a long, long time.